trio/testing/_fake_net.py

```diff
@@ -352,31 +377,56 @@ def getsockname(self):
             assert self.family == trio.socket.AF_INET6
             return ("::", 0)
 
-    def getpeername(self):
+    def getpeername(self) -> None:
         self._check_closed()
         if self._binding is not None:
             if self._binding.remote is not None:
```

Comment:
> trio/testing/_fake_net.py:(383:16 - 383:35): "UDPBinding" has no attribute "remote"  [attr-defined]

this method is never run in CI, so I suspect it's the code rather than the type hints that are wrong?

Reply:
added a bunch of asserts that will hopefully elucidate anybody trying to use this about the weirdness